### ISSUE TITLE
`IDE0059` is not fired for property pattern assignment

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#66573]
**Version Used**: 
17.5 Preview 4

**Steps to Reproduce**:
Case1:
```cs
object obj = "str";

if (obj is string str)
{

}
```
Case 2:
```cs
object obj = "str";

if (obj is string { Length: > 0 } str)
{

}
```
Case 3:
```cs
object obj = "str";

if (obj is string { } str)
{

}
```

**Diagnostic Id**:
IDE0059

**Expected Behavior**:
`IDE0059` is reported for all 3 cases

**Actual Behavior**:
`IDE0059` is reported only for case 1

**Comment by Youssef1313** [comment]
@kinkeadian25

**Comment by kinkeadian25** [comment]
I will work on this, thanks @Youssef1313 !

**Comment by kinkeadian25** [comment]
When I plug the first example into Sharplab, it says it is Declaration Pattern. The other two it gives me property pattern. So, does that mean it is not covering Property Pattern at any point as of now?

**Comment by Youssef1313** [comment]
@kinkeadian25 Not sure, but generally, the analyzer might not be dealing with syntax. So you'd need to debug a unit test to see when things go wrong.
